src/server/rdb_load.cc

```diff
@@ -1524,7 +1525,8 @@ struct RdbLoader::ObjSettings {
   ObjSettings() = default;
 };
 
-RdbLoader::RdbLoader(ScriptMgr* script_mgr) : script_mgr_(script_mgr) {
+RdbLoader::RdbLoader(Service* service)
+    : service_{service}, script_mgr_{service == nullptr ? nullptr : service->script_mgr()} {
   shard_buf_.reset(new ItemsBuf[shard_set->size()]);
 }
```

Comment:
This looks very tricky, maybe we should pass both explicitly?

Reply:
what looks tricky?

Reply:
Extracting and using the script manager under the hood